### PR TITLE
Map lifting and change permission handling

### PR DESCRIPTION
**Pull request by ifaresse**
## 🔄 Changes Overview

This PR focuses on improving location functionality, optimizing code, and updating tests to reflect recent changes.

---

### Location Enhancements
- **Reduced Latency**: Improved map performance by removing latency for accessing the user's location.
- **Permission Handling**: Changed the workflow for requesting location permissions, streamlining the process.

---

### Code Cleanup
- Removed unnecessary functions across multiple files.
- Updated and optimized logs for better clarity and debugging.
- Removed `PreviewSettingsScreen` as it was deemed unnecessary.

---

### Map Tests
- Adapted existing map tests to align with the recent changes, ensuring consistency and reliability.

### CODE REVIEW

**Review by noezaabi — APPROVED** [review]
This pull request refactors the Android app's permission handling by introducing LocationPermission and NotificationsPermission classes, which improves code modularity and readability. It removes the mock Location object from MapScreenTest.kt and SimpleMapTest.kt, simplifying the test setup. In MainActivity.kt, separate launchers for location and notification permissions streamline the request process, and the notification channel creation is relocated to ensure proper initialization. The MapScreen and SimpleMap composables are simplified by removing the location parameter, relying instead on the userViewModel for location data. Logging in LocationUpdateWorker.kt is reduced, focusing on essential information. Overall, these changes enhance the code structure, making it more maintainable and easier to understand. Well done !

**Review by DanR2003 — APPROVED** [review]
- The reduction in map latency seems to visibly have an impact which is very good for the general flow and smoothness of the app.

- Cleaning up unused functions and removing PreviewSettingsScreen was performed correctly, without removing any other parts relevant for the project. 

- Good job on ensuring the tests reflect the changes too.

- However, some parts could be better documented.

Overall, a clean and robust update. Nice job! :)

File: app/src/main/java/com/swent/suddenbump/model/notifications/NotificationsPermission.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import android.Manifest
+import android.app.Activity
+import android.content.pm.PackageManager
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.core.content.ContextCompat
+
+class NotificationsPermission(val activity: Activity) {
+
```
[the remark this answers is not in the document]
Comment (ifaresse) [reply app/src/main/java/com/swent/suddenbump/model/notifications/NotificationsPermission.kt:11]: Yes thank you for the insight

**Review by DanR2003 — APPROVED** [review]
Nice to see you addressed what I asked for :)

**Pull request merged**